### PR TITLE
sarkars/Change inp options

### PR DESCRIPTION
**Pull request by sayantan-nervana**
Made some user-friendly input option changes to tf2ngraph

1. --outnodes is made "required"
2. Correct an assert message (the assert states one of the input options (pb/pbtxt/savedmodel) and one of the output options (pb/pbtxt/savedmodel) should be present.
3. Added some description text to the help option

### CODE REVIEW

**Comment by sayantan-nervana** [comment]
https://github.com/tensorflow/ngraph-bridge/blob/master/tools/tf2ngraph.py#L93

The description provides sample usage. ie, `python tf2ngraph.py -h`.

The help will print the following (along with help for options):
```
Tool to convert TF graph into a ngraph enabled graph
    Sample usage:
    Command line:
    python tf2ngraph.py --inputsavedmodel test_graph_SM --outnodes out_node --outputpbtxt test_graph_SM_mod.pbtxt
    python tf2ngraph.py --inputpbtxt test_graph_SM.pbtxt --outnodes out_node --outputpbtxt test_graph_SM_mod.pbtxt
    or:
    functional api
    from tf2ngraph import convert
    convert('savedmodel', 'test_graph_SM' , 'pbtxt', 'test_graph_SM_mod.pbtxt', ['out_node'])
    convert('pbtxt', 'test_graph_SM.pbtxt' , 'pbtxt', 'test_graph_SM_mod.pbtxt', ['out_node'])
```

**Comment by sayantan-nervana** [comment]
Adding slack review comments here as TODOs:

1. Please change the option names to snake case to be consistent with PEP as well as other TensorFlow cmd line option convention. For example, `--inputsavedmodel` --> `--input_saved_model`. Also replace `outnodes` to `output_nodes`. 
---->yes, can make this change. users will just have to aware of the change and update their scripts
- [ ]

2. Also - don’t mix command line API with functional API. Document the functional API in the API documentation.
---->Do you mean in `-h` command line prints, do not mention the functional API? sounds good
- [ ]

3. For command line, use actual names i.e., pick up a real example pb file, actual input layer name and an actual output layer name)
----> Do you mean some file like `https://github.com/tensorflow/ngraph-bridge/blob/master/test/test_axpy.pbtxt` thats present in ngraph-bridge repo? sounds good
- [ ]

Tagging @avijit-nervana for commenting

**Comment by avijit-nervana** [comment]
@sayantan-nervana For the 3rd comment, what I mean for the filename  is something like `inception_v3_2016_08_28_frozen..pb` - that's all.

File: tools/tf2ngraph.py [hunk 1]
```diff
@@ -96,24 +96,29 @@ def prepare_argparser(formats):
...
-    in_out_groups = [parser.add_argument_group(i) for i in ['input', 'output']]
+    in_out_groups = [
+        parser.add_argument_group(i, j) for i, j in zip(
+            ['input', 'output'], ['Input formats', 'Output formats'])
+    ]
     for grp in in_out_groups:
         inp_out_group = grp.add_mutually_exclusive_group()
         for format in formats[grp.title]:
-            opt_name = grp.title + format
+            opt_name = grp.title + '_' + format
```
Comment (sayantan-nervana) [inline tools/tf2ngraph.py]: makes `inputpbtxt`-->`input_pbtxt`

File: tools/tf2ngraph.py [hunk 2]
```diff
@@ -96,24 +96,29 @@ def prepare_argparser(formats):
...
             inp_out_group.add_argument(
                 "--" + opt_name, help="Location of " + grp.title + " " + format)
     # Note: no other option must begin with "input" or "output"
     parser.add_argument(
-        "--outnodes", help="Comma separated list of output nodes")
+        "--output_nodes",
+        help=
+        "Comma separated list of output nodes. Output nodes can be found " \
+        "by manual inspection of the graph, prior knowledge or running the " \
+        "summarize_graph tool provided by Tensorflow",
```
Comment (sayantan-nervana) [inline tools/tf2ngraph.py]: Added some info about how one might identify `output_nodes`

File: tools/tf2ngraph.py [hunk 3]
```diff
@@ -96,24 +96,29 @@ def prepare_argparser(formats):
...
         inp_out_group = grp.add_mutually_exclusive_group()
         for format in formats[grp.title]:
-            opt_name = grp.title + format
+            opt_name = grp.title + '_' + format
             inp_out_group.add_argument(
                 "--" + opt_name, help="Location of " + grp.title + " " + format)
     # Note: no other option must begin with "input" or "output"
     parser.add_argument(
-        "--outnodes", help="Comma separated list of output nodes")
+        "--output_nodes",
```
Comment (sayantan-nervana) [inline tools/tf2ngraph.py]: `outnodes`->`output_nodes`

File: tools/tf2ngraph.py [hunk 4]
```diff
@@ -96,24 +96,29 @@ def prepare_argparser(formats):
...
-    python tf2ngraph.py --inputpbtxt test_graph_SM.pbtxt --outnodes out_node --outputpbtxt test_graph_SM_mod.pbtxt
-    or:
-    functional api
-    from tf2ngraph import convert
-    convert('savedmodel', 'test_graph_SM' , 'pbtxt', 'test_graph_SM_mod.pbtxt', ['out_node'])
-    convert('pbtxt', 'test_graph_SM.pbtxt' , 'pbtxt', 'test_graph_SM_mod.pbtxt', ['out_node'])
+    python tf2ngraph.py --input_savedmodel resnet_model_location --output_nodes out_node --output_pbtxt resnet_ngraph.pbtxt
+    python tf2ngraph.py --input_pbtxt mobilenet.pbtxt --output_nodes out_node --output_pbtxt mobilenet_ngraph.pbtxt
+    python tf2ngraph.py --input_pb inception_v3_2016_08_28_frozen.pb --output_nodes InceptionV3/Predictions/Reshape_1 --output_pb inception_v3_2016_08_28_frozen_ngraph.pb
+    python tf2ngraph.py --input_pbtxt ../test/test_axpy.pbtxt --output_nodes add --output_pbtxt axpy_ngraph.pbtxt
```
Comment (sayantan-nervana) [inline tools/tf2ngraph.py]: Added a "ready to run" command line using a pbtxt from our repo

File: tools/tf2ngraph.py [hunk 5]
```diff
@@ -96,24 +96,29 @@ def prepare_argparser(formats):
...
-    python tf2ngraph.py --inputsavedmodel test_graph_SM --outnodes out_node --outputpbtxt test_graph_SM_mod.pbtxt
-    python tf2ngraph.py --inputpbtxt test_graph_SM.pbtxt --outnodes out_node --outputpbtxt test_graph_SM_mod.pbtxt
-    or:
-    functional api
-    from tf2ngraph import convert
-    convert('savedmodel', 'test_graph_SM' , 'pbtxt', 'test_graph_SM_mod.pbtxt', ['out_node'])
-    convert('pbtxt', 'test_graph_SM.pbtxt' , 'pbtxt', 'test_graph_SM_mod.pbtxt', ['out_node'])
+    python tf2ngraph.py --input_savedmodel resnet_model_location --output_nodes out_node --output_pbtxt resnet_ngraph.pbtxt
+    python tf2ngraph.py --input_pbtxt mobilenet.pbtxt --output_nodes out_node --output_pbtxt mobilenet_ngraph.pbtxt
+    python tf2ngraph.py --input_pb inception_v3_2016_08_28_frozen.pb --output_nodes InceptionV3/Predictions/Reshape_1 --output_pb inception_v3_2016_08_28_frozen_ngraph.pb
```
Comment (sayantan-nervana) [inline tools/tf2ngraph.py]: Added a command line with inception and its actual output node name

**Pull request merged**